db/migrate/20181025230931_add_comments_to_vms_table.rb

```diff
@@ -0,0 +1,81 @@
+    change_column_comment :vms, :location, "The location or region of the VM."
+    change_column_comment :vms, :memory_hot_add_enabled, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :memory_hot_add_increment, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :memory_hot_add_limit, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :memory_limit, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :memory_reserve, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :memory_reserve_expand, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :memory_shares, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :memory_shares_level, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :miq_group_id, "MIQ Group associated with the VM."
+    change_column_comment :vms, :name, "The name of the VM/Template."
+    change_column_comment :vms, :orchestration_stack_id, "ID of the orchestration stack associated with the VM."
+    change_column_comment :vms, :power_state, "The current power state, typically 'on' or 'off'."
+    change_column_comment :vms, :previous_state, "The previous power state change."
+    change_column_comment :vms, :publicly_available, "Indicates whether the VM is public or private."
+    change_column_comment :vms, :raw_power_state, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :registered, "Indicates whether the VM registered to a Provider."
+    change_column_comment :vms, :resource_group_id, "ID of the resource group that the VM/Template belongs to."
+    change_column_comment :vms, :retired, "Indicates whether or not the VM is retired."
+    change_column_comment :vms, :retirement_last_warn, "Date of last retirement warning."
+    change_column_comment :vms, :retirement_requester, "Userid of the user who requested retirement."
+    change_column_comment :vms, :retirement_state, "The current retirement state."
+    change_column_comment :vms, :retirement_warn, "Warning message when VM was put into retirement."
+    change_column_comment :vms, :retires_on, "Date the VM is to be retired."
+    change_column_comment :vms, :smart, "[DEFINITION NEEDED]"
+    change_column_comment :vms, :standby_action, "Action taken when VM is put into standby mode."
+    change_column_comment :vms, :storage_id, "ID of the storage object associated with the VM."
+    change_column_comment :vms, :storage_profile_id, "ID of the storage profile associated with the VM."
+    change_column_comment :vms, :state_changed_on, "The date of the last power state change."
+    change_column_comment :vms, :template, "Indicates if a template was used to provision the VM or not."
+    change_column_comment :vms, :tenant_id, "ID of the tenant associated with the VM."
+    change_column_comment :vms, :tools_status, "Status of VMWare tools."
+    change_column_comment :vms, :type, "An internal class name that includes the provider type."
+    change_column_comment :vms, :uid_ems, "[DEFINITION NEEDED]"
```

Comment:
The globally unique reference used to reconnect VMs disconnected from other providers.  Most providers set this equal to the ems_ref (either because most have globally unique ems_refs anyway, or because there isn't anything better available).

For vmware this is the bios uuid which comes from the vmx file, so if you move this vm to another vcenter the uid_ems will stay the same even if the ems_ref is different.